### PR TITLE
[fix](delete) Incorrect precision detection for the decimal type in condition.​

### PR DESCRIPTION
**Pull request by mrhhsg**
## Proposed changes

For precision like Decimal(7,7),  the value "0.1234567" should be valid(the integer part is 0).

### CODE REVIEW

**Comment by mrhhsg** [comment]
run buildall

File: regression-test/suites/delete_p0/test_delete.groovy [hunk 1]
```diff
@@ -535,13 +536,18 @@ suite("test_delete") {
...
+        (true, '-1', 10, 0.7654321);
     """
 
     sql """
         delete from table_decimal where k1 = false and k2 = '-9999782574499444.2' and k3 = '-20';
     """
+
+    sql """
+        delete from table_decimal where k4 = '0.1234567';
+    """
```
Comment (yiguolei) [inline regression-test/suites/delete_p0/test_delete.groovy:549]: add test for  -0.123

**Comment by mrhhsg** [comment]
run buildall

**Comment by mrhhsg** [comment]
run buildall

**Comment by yiguolei** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]

**Review by Gabriel39 — APPROVED** [review]

**Review by jacktengg — APPROVED** [review]

**Pull request merged**